src/main/java/games/strategy/net/nio/Encoder.java

```diff
@@ -37,21 +37,20 @@ void write(final SocketChannel to, final MessageHeader header) {
     if (to == null) {
       throw new IllegalArgumentException("No to channel!");
     }
-    final ByteArrayOutputStream2 sink = new ByteArrayOutputStream2(512);
-    SocketWriteData data;
     try {
+      final ByteArrayOutputStream sink = new ByteArrayOutputStream(512);
       write(header, objectStreamFactory.create(sink), to);
-      data = new SocketWriteData(sink.getBuffer(), sink.size());
+      final SocketWriteData data = new SocketWriteData(sink.toByteArray(), sink.size());
```

Comment:
Here's the essential change: `sink.getBuffer()` was changed to `sink.toByteArray()`